esercizio_carcere/index.js

```diff
@@ -0,0 +1,90 @@
+
+
+
+
+document.addEventListener('DOMContentLoaded', async ()=>{
+
+    const usersResponse = await fetch('http://dms.cyberdelia.eu/api/v1/user');
+    const usersRaw = await usersResponse.json();
+    // const users = usersRaw.filter((user) => {
+    //     return user.firstname && user.lastname;
+    // });
+
+    //destructuring 
+    // const users = usersRaw.filter(({ firstname, lastname}) => {
+    //     return firstname && lastname;
+    // });
+
+    //instant return arrow function
+    let usersInPrison = usersRaw.filter(({ firstname, lastname}) => (firstname && lastname) );
+    let freeUsers = [];
+    const prison = document.getElementById('prison');
+    const freedom = document.getElementById('freedom');
+    
+
+
+    const judgeAction = (action, user) => {
+        if(action == 'Scarcera'){
+            usersInPrison = usersInPrison.filter((u) => {
+                return user.id != u.id;
+            });
+            freeUsers.push(user);
+            populate(usersInPrison, action, prison);
+            action = 'Incarcera';
+            populate(freeUsers, action, freedom);
+            return; 
+        }
+        if(action == 'Incarcera'){
+            freeUsers = freeUsers.filter((u) => {
+                return user.id != u.id;
+            });
+            usersInPrison.push(user);
+            populate(freeUsers, action, freedom);
+            action = 'Scarcera';
+            populate(usersInPrison, action, prison);
```

Comment:
Non è una best practice riassegnare un parametro, soprattutto in questo caso perchè non ti è necessario farlo
```suggestion
            action = 'Scarcera';
            populate(usersInPrison, "scarcera", prison);
```